Assets/MixedRealityToolkit/Common/Scripts/ClipPlane.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace MixedRealityToolkit.Common
+{
+    /// <summary>
+    /// Utility component to animate and visualize a clipping plane that can be used with 
+    /// the "MixedRealityToolkit/Standard" shader "_ClippingPlane" feature.
+    /// </summary>
+    [ExecuteInEditMode]
+    public class ClipPlane : MonoBehaviour
+    {
+        public Renderer[] Renderers;
+
+        private int clipPlaneID;
+        private MaterialPropertyBlock materialPropertyBlock;
+
+        private void Awake()
+        {
+            Initialize();
+            UpdatePlanePosition();
+        }
+
+#if UNITY_EDITOR
```

Comment:
Just to make things a bit cleaner we could use `Applicaiton.isEditor` to check this and remove the compiler symbols.

Reply:
That would incur extra Update calls at runtime. But, I agree readability is probably paramount here.

Reply:
Update gets called anyway on all MonoBehaviours, so it's really not a big deal.

Reply:
I don't believe that is true, check out the topic "How Update is called" in this blog post: https://blogs.unity3d.com/2015/12/23/1k-update-calls/ 

But, there is probably not going to be many of these behaviors in a scene, so it's a moot optimization, I'll make it more readable!

Reply:
Oh that's right, they did fix that.

I've been using Unity for too long haha.  That was a good blog post too.